### PR TITLE
Add type definition for pipeP and tryCatch

### PR DESCRIPTION
**Pull request by raychan**
@briancavalier I added our local change for ramda to flow-type
I am wondering if there is better way to test?

### CODE REVIEW

File: definitions/npm/ramda_v0.x.x/flow_v0.39.x-/ramda_v0.x.x.js
Comment (briancavalier) [inline definitions/npm/ramda_v0.x.x/flow_v0.39.x-/ramda_v0.x.x.js]: We could reuse `UnaryFn` here:

```js
declare type UnaryPromiseFn<A,R> = UnaryFn<A, Promise<R>>
```

It seems like we could do that to match the style of `UnarySameTypeFn` and `BinarySameTypeFn` below, which reuse `UnaryFn` and `BinaryFn`.

File: definitions/npm/ramda_v0.x.x/flow_v0.39.x-/ramda_v0.x.x.js
Comment (briancavalier) [inline definitions/npm/ramda_v0.x.x/flow_v0.39.x-/ramda_v0.x.x.js:132]: I don't fully understand why they've chosen to use one style of declarations (intersection type plus `declare var`) for certain curried functions, like `pipe` and `pipeP`, and a different style of declarations (`declare function` variants) for other curried functions, like `tryCatch`.  Do you know?

It seems like the right thing to match the existing style, though, like you did :+1:

File: definitions/npm/ramda_v0.x.x/flow_v0.39.x-/ramda_v0.x.x.js
Comment (briancavalier) [inline definitions/npm/ramda_v0.x.x/flow_v0.39.x-/ramda_v0.x.x.js:717]: Why use an intersection type to define the 3rd and 4th curried variants here instead of just adding a 4th `declare function`?

File: definitions/npm/ramda_v0.x.x/flow_v0.39.x-/test_ramda_v0.x.x_functions.js
Comment (briancavalier) [inline definitions/npm/ramda_v0.x.x/flow_v0.39.x-/test_ramda_v0.x.x_functions.js]: This case, tc4, *seems* like it's the same as tc2 + tc3.  Is there a reason to have both?

File: definitions/npm/ramda_v0.x.x/flow_v0.39.x-/test_ramda_v0.x.x_functions.js
Comment (briancavalier) [inline definitions/npm/ramda_v0.x.x/flow_v0.39.x-/test_ramda_v0.x.x_functions.js:63]: Nice catch :+1:

**Review by briancavalier — CHANGES_REQUESTED** [review]
Thanks!  Looks good--just a few questions.  Hopefully, once we merge this to master, we can PR it from our fork to flowtype/flow-typed.

File: definitions/npm/ramda_v0.x.x/flow_v0.39.x-/ramda_v0.x.x.js
Comment (raychan) [reply definitions/npm/ramda_v0.x.x/flow_v0.39.x-/ramda_v0.x.x.js:132]: I don't, I am wondering too. Maybe different people contribute to this?

File: definitions/npm/ramda_v0.x.x/flow_v0.39.x-/ramda_v0.x.x.js
Comment (raychan) [reply definitions/npm/ramda_v0.x.x/flow_v0.39.x-/ramda_v0.x.x.js:717]: I spent so much time to figure this out. I originally had 4 `declare function`, but test `tc5` or `tc6` will break. Only the 3rd declaration will be used.

File: definitions/npm/ramda_v0.x.x/flow_v0.39.x-/test_ramda_v0.x.x_functions.js
Comment (raychan) [reply definitions/npm/ramda_v0.x.x/flow_v0.39.x-/test_ramda_v0.x.x_functions.js]: true, I will delete it

**Comment by raychan** [comment]
> Hopefully, once we merge this to master, we can PR it from our fork to flowtype/flow-typed.

Yes, I am pretty sure we can. Or we can just create a PR from this branch.

**Comment by raychan** [comment]
I am wondering if we need to update for older flow versions?

**Comment by raychan** [comment]
@briancavalier  Any more comments for this PR?

File: definitions/npm/ramda_v0.x.x/flow_v0.39.x-/ramda_v0.x.x.js
Comment (briancavalier) [reply definitions/npm/ramda_v0.x.x/flow_v0.39.x-/ramda_v0.x.x.js:132]: Yeah, that's possible.  Seems like the maintainers would want to maintain a consistent style though.  Oh well, as long as we match the existing style, we should be ok.

File: definitions/npm/ramda_v0.x.x/flow_v0.39.x-/ramda_v0.x.x.js
Comment (briancavalier) [reply definitions/npm/ramda_v0.x.x/flow_v0.39.x-/ramda_v0.x.x.js:717]: That's really strange.  I'll play around with it a bit.

**Comment by briancavalier** [comment]
> I am wondering if we need to update for older flow versions?

We probably don't, but when we submit a PR to flow-typed, they may care.  So, we could do it now, or wait until we submit a PR.

File: definitions/npm/ramda_v0.x.x/flow_v0.39.x-/ramda_v0.x.x.js
Comment (briancavalier) [reply definitions/npm/ramda_v0.x.x/flow_v0.39.x-/ramda_v0.x.x.js:717]: Hmmm, I wonder if the order of the declarations matters for some reason ...

Comment (raychan) [reply definitions/npm/ramda_v0.x.x/flow_v0.39.x-/ramda_v0.x.x.js:717]: 😕

**Comment by raychan** [comment]
ok, let's wait then

**Review by briancavalier — APPROVED** [review]
Nice work!

**Pull request merged**
